blob/fileblob/fileblob_test.go

```diff
@@ -33,27 +34,62 @@ import (
 )
 
 type harness struct {
-	dir    string
-	closer func()
+	dir        string
+	server     *httptest.Server
+	urlSignKey string
+	closer     func()
 }
 
 func newHarness(ctx context.Context, t *testing.T) (drivertest.Harness, error) {
 	dir := path.Join(os.TempDir(), "go-cloud-fileblob")
 	if err := os.MkdirAll(dir, os.ModePerm); err != nil {
 		return nil, err
 	}
-	return &harness{
-		dir:    dir,
-		closer: func() { _ = os.RemoveAll(dir) },
-	}, nil
+
+	h := &harness{dir: dir, urlSignKey: "I'm a secret key"}
+	localServer := httptest.NewServer(http.HandlerFunc(h.serveSignedURL))
+	h.server = localServer
+	h.closer = func() { _ = os.RemoveAll(dir); localServer.Close() }
+
+	return h, nil
+}
+
+func (h *harness) serveSignedURL(w http.ResponseWriter, r *http.Request) {
+	// URLSignerHMAC requires the scheme, host, and path to construct the
+	// signed URL, but validating only requires the key.
+	urlSigner := NewURLSignerHMAC("", "", "", h.urlSignKey)
```

Comment:
Could you save `urlSigner` on the harness instead of recreating it here?

Reply:
I could, but I thought it might be better to show it being re-created with the signing key, since I doubt that people using it in their applications would have access to the same in-memory object.

Reply:
My guess it that people are just as likely to have the original `urlSigner` around as they are to have the key. They will also definitely want to have the original `blob.Bucket` instead of creating a new one as well.